### ISSUE TITLE
Code and Javadoc cleanup

### ISSUE THREAD

**Issue by milczarekIT** [ta4j/ta4j#426]
I created PR https://github.com/ta4j/ta4j/pull/425 with lots of Javadoc and code clean ups.

I noticed that in Javadoc there are a lot of ways of new lines including:
- just new line
- `<p>`
- `<p/>`
- `</p>`
- `<p></p>`

So I unified them to just new line. This format is used also in license header.
To javadocs was also added links to documentation for indicators (mainly to investopedia)

I cleaned up some code smell in code like:
- upper case names for non static variables
- removed `_` from non static variables
- and much more

Purpose of mentioned PR is to increase code quality.

Some time ago I raised issue 'Common code formatter for project' (https://github.com/ta4j/ta4j/issues/320), but without results. In my opinion we should rethink using common code formatter.

**Comment by team172011** [comment]
Maybe we should think about how to implement solutions provided in #320 . If we could have an IDE independent maven plugin that would be great. I agree that a clean code base is important but I also noticed that such "clean code" and "added javadoc to hundreds of files" PRs like your #425 are made frequently by different users, each having their own understaning of how this could be realized. PRs that are touching so many files but do not add any new functionality to the lib should be avoided as well as possible.

**Comment by nimo23** [comment]
>PRs that are touching so many files but do not add any new functionality to the lib should be avoided as well as possible.

I find @team172011 is right. Such PRs **does not provide any added values**. It is okay, if one **provides a value** and clean the code **by the way.** But only cleaning the code because of subjective preferences should be avoided: There are common code styles every developers should consider. But _code styles_ are also subjective, such should be avoided to add or exaggerate restrictive code styles only because one thinks this is the "best" code style or the "best practice". "Best practice" is which solves issues.. Developers have different meanings and I prefer contributers adding values to this lib instead of preventing to do so because they fear about such exaggerated code style "barriers".

**Comment by milczarekIT** [comment]
> like your #425 are made frequently by different users, **each having their own understaning** of how this could be realized.

Yes, so this is reason why this pull request was created. I noticed that each contributor has own  fortmatting rules, so I found which one is the most popular for given 'issue' and I just unified it.

I was thinking that is was clear after my clarifications:

> I noticed that in Javadoc there are a lot of ways of new lines including:
just new line
`<p>`
`<p/>`
`</p>`
`<p></p>`

I was pretty sure that is was clear, because I added additional explanations

> So I unified them to just new line. This format is used also in license header.

In other words: I cleaned up mess in javadocs

I'm worried about this library... if contributors do not want to fix mess in code/documentation.

I think it is easy way to remove problem with incorrect usage of javadoc tags, poor documentation
1. I fixed smells in javadocs (also issues with incorrect usage of `@see` tag, missing links to reference article e.g. at Investopedia). 
2. If someone find other, they shuold be fixed too.
3. Since now during code review also javados should be reviewed, so unified way of adding new lines or correct usage of `@see` tag should be also checked

> PRs that are touching so many files but do not add any new functionality to the lib should be avoided as well as possible.

I agree with that, but it will be possible in the future when we define: Definition of Done, Code of conduct, Code formatter and ... we will clean all mess in sources.
I am trying to do it step by step..., but I'm worried about this library 


What is more I removed code smells like:

> upper case names for non static variables
removed _ from non static variables
and much more

which should never go to develop branch. So we need PRs like this
